### ISSUE TITLE
Attempting NOOP on dropped connection

### ISSUE THREAD

**Issue by sven-strothoff** [OfflineIMAP/offlineimap#69]
I am running OfflineIMAP 6.5.5 on OS X. I have launchd configured to start OfflineIMAP in the background. I sync a single IMAP mailbox to a local MailDir.
OfflineIMAP is working fine for some time, but then no more mails arrive. If I take a look at the logs the last entries are always (except the thread numbers of course):

```
Thread-655:
 Attempting NOOP on dropped connection imap.zoho.com
Thread-657:
 Attempting NOOP on dropped connection imap.zoho.com
Thread-656:
 Attempting NOOP on dropped connection imap.zoho.com
```

I suspect that it happens when the computer was entering sleep state or the internet connection dropped. Even though OfflineIMAP is still running it never recovers from the connection loss and I have to kill it and start it again.

At first I was using IDLE, but after I read that it is still experimental I thought it might be he culprit here, but the behaviour is still the same after dialling IDLE.

I found some discussion of the NOOP message on some mailing list archive, but that was several years old, so I am not sure if it was ever resolved or if this is a new issue.

If you tell me what you need I am happy to add additional information or debug logs. I would gladly help in resolving this issue, however I have no idea where to start looking on my own…

**Comment by nicolas33** [comment]
We might still want to close offlineimap after n NOOP rather than looping indefinitely.

**Comment by quite** [comment]
Yeah. I'm living with a cronjob uses netstat to see if the connection to my
imap server is alive, and a -9 if it's not. But that's OK too. : )

**Comment by sc68cal** [comment]
I'm highly mobile with my laptop, moving between WiFi networks (commute between home and office) - and I had `offlineimap` running via `launchd` and basically just `pkill -9 -f offlineimap` whenever it misbehaves. 

```
 *** Finished account 'personal' in 0:02
 Next refresh in 5.0 minutes
 Next refresh in 4.0 minutes
 Next refresh in 3.0 minutes
 Next refresh in 2.0 minutes
 Next refresh in 1.0 minutes
 *** Processing account personal
```

**Comment by andreas-bulling** [comment]
I am using the current version (7.0.14) on OSX and have the same problem. Playing around with different values for `socktimeout`, `holdconnectionopen` and `keepalive` didn't help (not even sure if these parameters are related at all to this issue).

**Comment by andreas-bulling** [comment]
Switching to mbsync (http://isync.sourceforge.net/mbsync.html) solved all problems.

**Comment by chris001** [comment]
Switch keepalive to a smaller value such as 60 or 30 fixes the issue of not sending keepalives.
For the case of the internet connection dropped or imap connection dropped - this needs to be detected and handled - restart the sync procedure, reconnect to imap.
